qemu/tests/fio_perf.py

```diff
@@ -0,0 +1,224 @@
+    _pin_vm_threads(vm, params.get("numa_node"))
+
+    # get parameter from dictionary
+    fio_cmd = params.get('fio_cmd')
+    write_cmd = params.get('write_cmd')
+    rw = params.get("rw")
+    block_size = params.get("block_size")
+    iodepth = params.get("iodepth")
+    threads = params.get("threads")
+    cmd_timeout = int(params.get("cmd_timeout", 1200))
+    order_list = params.get("order_list")
+    driver_format = params.get("drive_format")
+    guest_ver_cmd = params.get("guest_ver_cmd")
+    pattern = params.get("pattern")
+    pre_cmd = params.get("pre_cmd")
+    format = params.get("format")
+    md5value = params.get("md5value")
+    compile_cmd = params.get("compile_cmd")
+    test_src = params.get("test_src")
+    os_type = params.get("os_type")
+    image_format = params.get("image_format")
+    drop_cache = params.get("drop_cache")
+
+    result_path = utils_misc.get_path(test.resultsdir,
+                                      "fio_result.RHS")
+    result_file = open(result_path, "w")
+
+    # scratch host and windows guest version info
+    get_version(session, result_file, guest_ver_cmd, os_type,
+                driver_format, cmd_timeout)
+
+    # online disk
+    if os_type == "windows":
+        for num in range(1, 2):
+            disks = check_status(session, cmd_timeout, num)
+            diskstatus = re.findall("Disk\s+\d+\s+(\w+).*?\s+\d+", disks[0])[0]
+            if diskstatus == "Offline":
+                online_disk_cmd = params.get("online_disk_cmd")
+                online_disk_run = online_disk_cmd % num
+                (s, o) = session.cmd_status_output(online_disk_run,
+                                                   timeout=cmd_timeout)
+                if s != 0:
+                    raise error.TestFail("Failed to online disk: %s" % o)
+    # format disk
+    if format == "True":
+        session.cmd(pre_cmd, cmd_timeout)
+
+    # get order_list
+    order_line = ""
+    for order in order_list.split():
+        order_line += "%s|" % format_result(order)
+
+    # get result tested by each seanario
+    for i in rw.split():
+        result_file.write("Category:%s\n" % i)
+        result_file.write("%s\n" % order_line.rstrip("|"))
+        if i == "rewrite":
+            i = "write"
+        for j in block_size.split():
+            for k in iodepth.split():
+                for l in threads.split():
+                    line = ""
+                    line += "%s|" % format_result(j[:-1])
+                    line += "%s|" % format_result(k)
+                    line += "%s|" % format_result(l)
+                    if format == "True":
+                        m = i + "_" + j + "_" + k
+                        run_cmd = fio_cmd % (i, j, k, m, l)
+                    else:
+                        run_cmd = fio_cmd % (i, j, k, l)
+
+                    logging.info("run_cmd is: %s" % run_cmd)
+                    if os_type == "linux":
+                        commands.getoutput(drop_cache)
```

Comment:
There was something wrong here, I want to execuate the drop_cache cmd in guest, not host, so I have changed the code.